agent/s3.go

```diff
@@ -74,6 +75,14 @@ func awsS3Session(region string) (*session.Session, error) {
 	return sess, nil
 }
 
+func sharedCredentialsProvider() credentials.Provider {
+	// If empty SDK will default to environment variable "AWS_PROFILE"
+	// or "default" if environment variable is also not set.
+	awsProfile := os.Getenv("BUILDKITE_S3_PROFILE")
+
+	return &credentials.SharedCredentialsProvider{Filename: "", Profile: awsProfile}
```

Comment:
[super nit] I think having the Filename explicitly blank here is a little confusing, even though that's what it defaults(?) to

```suggestion
	return &credentials.SharedCredentialsProvider{Profile: awsProfile}
```

Reply:
I tend to agree, I copied this code from the AWS SDK, but let's ditch it.